system-design/app-interoperability/application-package-definition.md

```diff
@@ -1,14 +1,25 @@
 # Application Package Definition
 
-This section addresses how an application is packaged by the “App Developer” who has implemented the application and aims to provide it to Margo-conformant systems. An application aggregates one or multiple [OCI Containers](https://github.com/opencontainers). The **application package** is made available in an [application registry](./workload-orch-to-app-reg-interaction.md).  
+This section addresses how an application is packaged by the “Application Developer” who has implemented the application and aims to provide it to Margo-conformant systems. An application aggregates one or more [OCI Containers](https://github.com/opencontainers). The **application package** is made available in an [application registry](./workload-orch-to-app-reg-interaction.md).  
 
-The application package comprises (a) the **application description** file `margo.yaml`, which contains informative contents and metadata (e.g., for application discovery purposes), (b) the **application manifests** ([Helm Chart](https://helm.sh/docs/) and [Docker Compose](https://github.com/compose-spec/compose-spec/blob/master/03-compose-file.md)) containing deployment instructions, and optionally (3) **application resources** (e.g., container binaries), which are necessary to enable the lifecycle of the application.
+The application package comprises:
 
-The application manifest SHALL be defined as a Helm Chart AND/OR a Docker Compose. If either one cannot be implemented it MAY be omitted.  
+- The **application description** file `margo.yaml`, which contains information about the application's marketing details (e.g., description, icon, release notes, license file, etc.), resource requirements, required input parameters, and application's installation packages (e.g,  Helm charts, docker-compose package) location.
+- The **application resources** which can be used to display additional information about the application in an application catalog or marketplace based on the application's defined metadata (e.g., description, icon, release notes, license file, etc.).
 
-Margo RECOMMENDS defining both (Helm Chart **AND** Docker Compose) for strengthening interoperability and applicability.  
+The application package components SHALL be defined as Helm Charts AND/OR a Docker Compose packages.
 
-A device that runs the application can only actively run either Docker Compose or Helm Chart.
+- To target devices running Kubernetes applications must be packaged as helm charts using [Helm V3](https://helm.sh/).
+- To target devices deploying applications using Docker Compose files you must create a tarball file containing the application's docker-compose.yml file and any additional artifacts referenced by the docker compose file (e.g., configuration files, environment variable files, etc.). It is highly recommend to digitally sign this package. When digitally signing the package PGP MUST be used.
```

Comment:
Did we already clarify whether we have to restrict the supported docker compose file format versions? There exist deprecations in the format and it might be that they removed in upcoming versions.

Reply:
They write:

*f some fields are unknown, typically because the Compose file was written with fields defined by a newer version of the Specification, you'll receive a warning message.*

For me it sounds, that deprecation might cause problems. What happens if such warning is issued? Are they backward compatible or not? Is field considered or not? 

As we are dealing with a tool chain, probably nobody will read (and understand) such a warning. We could avoid this by pinning a version range of docker compose. Then, we would be on the safe side.

Reply:
I updated the **Investigation Needed** inline note to also include this as one of the things we need to identify for docker-compose deployments.